pkg/operator/observe_config.go

```diff
@@ -92,12 +90,43 @@ func (c ConfigObserver) sync() error {
 	// don't worry about errors
 	currentConfig := map[string]interface{}{}
 	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&currentConfig)
-	if reflect.DeepEqual(currentConfig, observedConfig) {
-		return nil
+	observedConfig := map[string]interface{}{}
+	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&observedConfig)
+	currentStatus := operatorConfig.Status.DeepCopy()
+
+	var errs []error
+	for _, observer := range c.observers {
+		var currErrs []error
+		observedConfig, currErrs = observer(c.kubeClient, &rest.Config{}, observedConfig)
+		errs = append(errs, currErrs...)
 	}
 
-	glog.Infof("writing updated observedConfig: %v", diff.ObjectDiff(operatorConfig.Spec.ObservedConfig.Object, observedConfig))
-	operatorConfig.Spec.ObservedConfig = runtime.RawExtension{Object: &unstructured.Unstructured{Object: observedConfig}}
+	if len(errs) > 0 {
+		var messages []string
+		for _, currentError := range errs {
```

Comment:
@deads2k any particular reason we don't want to use `k8s.io/apimachinery/pkg/util/errors.Aggregate` and get rid of that loop?

Reply:
> @deads2k any particular reason we don't want to use `k8s.io/apimachinery/pkg/util/errors.Aggregate` and get rid of that loop?

It doesn't give me pretty newlines.